Sources/RequestDL/Internals/Sources/Session/Async Response/Internals.AsyncResponse.swift

```diff
@@ -40,28 +46,49 @@ extension Internals {
                     lastHead = head
                 }
 
-                self = .init(upload: nil, download: nil)
+                self = .init(
+                    uploadingBytes: uploadingBytes,
+                    upload: nil,
+                    download: nil
+                )
+
                 return lastHead.map {
-                    .download($0, .init(data))
+                    let totalSize = $0.headers["Content-Length"]?
+                        .reduce([]) { $0 + $1.split(separator: ",") }
+                        .lazy
+                        .map { $0.trimmingCharacters(in: .whitespaces) }
+                        .compactMap(Int.init)
+                        .max()
+
+                    return .download(DownloadStep(
+                        head: $0,
+                        bytes: AsyncBytes(
+                            totalSize: totalSize ?? .zero,
+                            stream: data
+                        )
+                    ))
```

Comment:
Since this chunk of code contains 4 ```$0```s, using a proper name to identify what you are handling would increase readability.

Reply:
You totally right. I will also move this code to an extension file, so we can use this in other places. Currently, this same code is already being used in others contexts.